tests/test_parser.py

```diff
@@ -0,0 +1,58 @@
+import types
+
+import pytest
+
+from nexus.parser import Nexus, TokenType
+
+
+@pytest.mark.parametrize(
+    'nex,expect,roundtrip',
+    [
+        (
+            """#NEXUS
+BEGIN TREES;
+TREE best = (fish , (frog,
+(snake, mouse))) ;
+END;""",
+            lambda n: sum(1 for t in n.nexus if t.type != TokenType.WHITESPACE) == 23,
+            True),
+        (
+            '#NEXUS begin trees; tree ; end;',
+            lambda n: n.blocks[0].name == 'TREES',
+            True),
+        (
+            "#NEXUS BEGIN AssuMP[co[mm]ent]TiONS; END;",
+            lambda n: n.blocks[0].name == 'ASSUMPTIONS' and n.comments[0].text == 'co[mm]ent',
+            False),  # We cannot roundtrip in-word comments!
+        # Commands cannot contain semicolons, except as terminators, unless the semicolons are
+        # contained within a comment or within a quoted token consisting of more than one text
+        # character.
+        (
+            "#NEXUS cmd [;] stuff;",
+            lambda n: n.commands[0].words[1].text == 'stuff',
+            True),
+        (
+            "#NEXUS cmd 'a;b' stuff;",
+            lambda n: n.commands[0].words[1].text == 'stuff',
+            True),
+        (
+            "#NEXUS cmd '[ab]' stuff;",
```

Comment:
Comments in quoted words are not parsed as comments.

Reply:
that makes sense to me